openslides_backend/action/actions/user/user_mixin.py

```diff
@@ -173,3 +173,55 @@ def get_history_information(self) -> Optional[HistoryInformation]:
                     fqid_from_collection_and_id("user", instance["id"])
                 ] = instance_information
         return information
+
+
+class DuplicateCheckMixin(Action):
+    def init_duplicate_set(
+        self, usernames: List[str], names_and_emails: List[Any]
+    ) -> None:
+        # we don't want to call "get_all", so need to construct this large
+        # filter here.
+        filter_ = Or(
+            *[FilterOperator("username", "=", un) for un in usernames],
+            *[
+                And(
+                    FilterOperator("first_name", "=", entry[0]),
+                    FilterOperator("last_name", "=", entry[1]),
+                    FilterOperator("email", "=", entry[2]),
+                )
+                for entry in names_and_emails
+            ],
+        )
+
+        users = self.datastore.filter(
+            "user",
+            filter_,
+            ["id", "username", "first_name", "last_name", "email"],
+            lock_result=False,
+        ).values()
```

Comment:
Is there a reason you don't call the `search_users` presenter here? Even if we want to filter case-sensitive here, adding a flag to the presenter instead of duplicating the code would be the way to go.

Reply:
I don't know, how to call a presenter in an action. I don't see the need to use the `search_users` here.

Reply:
Add a helper method analogous to `execute_other_action` to the base action class, but simplified since you don't need to care about created write requests and the like. You can call the presenter class like this: https://github.com/OpenSlides/openslides-backend/blob/88b6ea61d081e7c3e55256ec954f18fbd30c91fa/openslides_backend/presenter/presenter.py#L125.

There are two reasons to call the presenter, which go hand in hand:
- code deduplication
- one single place where the user matching algorithm is defined. We don't want the import to differ from the presenter in terms of user matching and would need to change both places everytime something about the algorithm changes.

Reply:
Okay, I added a helper `execute_presenter` in the action base class and use it to do the filtering.
I use the `SearchUsers` presenter in the context.